client/asset/interface.go

```diff
@@ -156,8 +156,8 @@ type Wallet interface {
 	// the registration fee, and the fee rate is taken from the DEX configuration.
 	PayFee(address string, feeAmt uint64) (Coin, error)
 	// Confirmations gets the number of confirmations for the specified coin ID.
-	// The ID need not represent an unspent coin, but coin IDs unknown to this
-	// wallet may return an error.
+	// If the coin is not unspent, and is not owned by this wallet,
+	// Confirmations may return an error.
 	Confirmations(ctx context.Context, id dex.Bytes) (uint32, error)
```

Comment:
I considered adding an `unspent bool` return value to the method too, but after examining the way the method is used in `Core`, I don't think it's necessary.

Reply:
Probably OK, but the recovery paths should be tested.   I'm also wondering if the isSwappable/isRedeemable checks might behave differently under certain circumstances if spent contracts are no longer a coin not found error.

Unfortunately the harness is still busted in a myriad of ways: https://github.com/decred/dcrdex/pull/799  We could at least change that test to use an encrypted btc wallet so the hack to block actions would work again.

Reply:
Also, while I there was a match recovery issue we need to look into in https://github.com/decred/dcrdex/issues/883#issuecomment-743417815, whatever happened there would have changed too because the audit would not have said "coin not found" (it was auditing a spent contract for some reason).  That issue still needs investigation despite being closed, but I'm noting it here because such an audit might have passed despite the contract being spent.

Reply:
Thoughts about use of `gettransaction` as fallback after `gettxout` in `asset.Wallet.Confirmations`: `getransaction` must be wallet transactions, which counterparty swaps shouldn't be, but asset backends might behave differently, or the contract could be made by counterparty to include an extra output that wallet recognizes thus changing the result.  The uncertainty of how this `Confirmations` method will behave for spent counterparty contracts is a concern.

---

`isSwappable` in `MakerSwapCast`
- As maker, will now notify confirmations after redeem and refund, until retire.  Still returns false though.
- As taker, could return `ready=true` even after the coin (maker's contract) is spent *depending on wallet capability* (getransaction), which in `MakerSwapCast` indicates maker refunded already.  This probably means revoked, which should make it return false before the confs check, but it's not guaranteed to be marked as revoked (e.g. server down) although not being revoked after maker contract expire is unlikely. Aside: taker swapping this late in the game would require taker making a contract that is already expired -- I think we've actually seen that happen, and we should be checking `lockTime` is not in the past in `swapMatchGroup`.

`isRedeemable` in `TakerSwapCast`
- As taker, will notify confirmations after redeem and refund, until retire.  Still returns false though.
- As maker, could say ready=true even after spent *depending on wallet capability* (getransaction), which in `TakerSwapCast` indicates taker refunded, which implies revoked, but `isRedeemable` **does not check `IsRevoked`** for sake of auto-redeem. (**Not good**) Thus, maker might try to redeem the refunded taker contract instead of going to refund their own by switching on `isRefundable`.  Unable to refund scenario.

`shouldBeginFindRedemption` as taker already used `wallet.Confirmations` to check own contract confirms. It will now hit `gettxout` first and will fall back to `gettransaction` when spent, so logging when spent might actually be useful since that's when you expect a redeem will be found right away.

The `t.notify` in `counterPartyConfirms` should be in the `if match.counterConfirms != int64(have)` condition.  Or is it necessary to (re)send this value to frontend on ticks and not just changes to the value?